File: python/unidad_02/problema_2_9.py
"""
El programa, dado como dato el precio básico de un artículo, calcula el
impuesto correspondiente del mismo teniendo en cuenta ciertos criterios.
"""

def aplicar_impuesto(precio):

    if precio <= 20:

        impuesto = 0

    elif precio <= 40:

        impuesto = (precio - 20) * 0.3

    elif precio <= 500:

        impuesto = 20 * 0.3 + (precio - 40) * 0.4

    else:

        impuesto = 20 * 0.3 + (precio - 40) * 0.5

    nuevo_precio = precio + impuesto

    return nuevo_precio

def main():

    try:

        precio = float(input("Precio del artículo: "))

        if precio <= 0:

            print("Error: el precio debe ser mayor que cero.")
            return
    
    except ValueError as e:

        print(e)
        return
    
    nuevo_precio = aplicar_impuesto(precio)

    print(f"El precio del artículo más impuestos")

    print(f"da un total de {nuevo_precio:.2f} u")

if __name__ == '__main__':

    main()

File: java/unidad_02/Problema_2_9/src/Problema_2_9.java
import java.util.Scanner;
import java.util.InputMismatchException;

/*
 * 
 * El programa, dado como dato el monto de la compra de un cliente, determina lo
 * que el mismo debe pagar teniendo en cuenta una serie de descuentos.
 *
 * @author Angel P.
 * 
 */

public class Problema_2_9 {

    public static float aplicarImpuesto(float precio) {

        float impuesto;

        if (precio <= 20) {

            impuesto = 0;

        } else if (precio <= 40) {

            impuesto = (precio - 20) * (float) 0.3;

        } else if (precio <= 500) {

            impuesto = 20 * (float) 0.3 + (precio - 40) * (float) 0.4;

        } else {

            impuesto = 20 * (float) 0.3 + (precio - 40) * (float) 0.5;
        }
        float nuevoPrecio = precio + impuesto;

        return nuevoPrecio;
    }

    public static void main(String[] args) {

        float precio;

        try (Scanner scanner = new Scanner(System.in)) {

            System.out.print("Precio del artículo: ");
            precio = scanner.nextFloat();

            if (precio <= 0) {

                System.out.println("Error: el precio debe ser mayor que cero.");
                return;
            }
        } catch (InputMismatchException e) {

            System.out.println(e);
            return;
        }
        float nuevoPrecio = aplicarImpuesto(precio);

        System.out.println("El precio del artículo más impuestos");

        System.out.printf("da un total de %.2f u", nuevoPrecio);
    }
}